Rework of the 'character' section

aphillips opened the pull request:
This implements discussion from the 2025-03-13 and 2025-03-20 teleconferences.

This adds a big note box at the start pointing to the encoding essentials article. But then we go on to define everything locally 😜 

- Make a new "basics" section at the top
- Remove most example boxes
- Reorder prose to flow better
- Define 'abstract character'
- Remove all 'meaningful' use of color
- Move the section "defining 'string'" just after the section on characters (and retitle for consistency)
- Replace the Devanagari example with "I love Swiss cows"
- Define 'encoded character'
- Add a table of "...instead of...use" terminology


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/aphillips/bp-i18n-specdev/pull/155.html" title="Last updated on Apr 10, 2025, 3:54 PM UTC (a1e267c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/bp-i18n-specdev/155/b063c1f...aphillips:a1e267c.html" title="Last updated on Apr 10, 2025, 3:54 PM UTC (a1e267c)">Diff</a>

aphillips committed:
Rewrite 'characters' section

- Add more feedback from @r12a, especially a11y issue, Devanagari code
  points, and a reduction is colored formatting
- Add a big note at the top sending people to essential concepts
- Move mustard to make it visible
- Provide better guidance about (not) using 'character'

aphillips committed:
Rewrite and reorder

- Remove the aside boxes and merge together
- Move defining string just after defining character

aphillips committed:
Radical rewrite in progress

aphillips committed:
Tightening definitions

aphillips committed:
Fix typo

aphillips committed:
Another typo. Sheesh.

aphillips committed:
Improve the terminology illustration

aphillips committed:
Fix start here styling, add UTF-8 illustration

aphillips committed:
Add UCD and some cleanup of character encoding

aphillips committed:
Made terminology illustration table clearer

xfq commented on index.html:
```diff
@@ -1261,213 +1262,203 @@ <h2>Characters</h2>
+    <legend>Start Here</legend>
+    <p>The relationship between text and its storage and processing in computers is complicated, involving many terms and concepts that may be new to you. Understanding this terminology is important to understanding the guidance in the internationalization guidelines and best practices. We've created this article to explain and illustrate what you need to know:</p>
+    
+    <p style="text-align:center;font-family:sans-serif;font-size:larger"><a href="https://www.w3.org/International/articles/definitions-characters/index.en.html#characters">Character encodings: Essential concepts</a></p>
```
```suggestion
    <p style="text-align:center;font-family:sans-serif;font-size:larger"><a href="https://www.w3.org/International/articles/definitions-characters/index.en.html">Character encodings: Essential concepts</a></p>
```

aphillips committed:
Address comments from 2025-03-20 telecon

- Fix link to essential concepts
- Reword the basic mustard item
- Reword the guidance in the other mustaard
- Merge mustard with text
- Adopt the D7 definition of 'abstract character' from Unicode
- Replace the Devanagari example with "I love Swiss cows" in (Russian)
  emoji
- The Oxford comma
- Very light edit of definition of string intro para

aphillips committed:
Add recommended term table, re-word mustard

aphillips committed:
Define `encoded character`

- Quote Unicode
- Link the use of the term in the 'string' section
- Clean up the \[Dx\] notation

aphillips committed:
Replace emoji sequence with graphic

aphillips committed:
Graphic for FE0F, lang for the cow text

aphillips commented:
@xfq @r12a I think this is nearing completion. Please give this a thorough review? I'd like to finish this project soon!

aphillips committed:
Tidy the 'instead of' table

aphillips committed:
Address comments, fix tables

aphillips committed:
Merge branch 'w3c:gh-pages' into character-intro

aphillips committed:
Fix broken family-example link

aphillips committed:
Add note about flag emoji display

aphillips commented:
Per the 2025-04-17 teleconference, merging this work. Since other work in this section is pending, additional changes can be made in a new PR.
